OISF/suricata
# Pcre2 3194 v11

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194

Describe changes:
- migrate all pcre1 code to pcre2
- fuzz: builds fuzz_sigpcap_aware only when asked (ie ./configure --enable-fuzztargets)
- qa : remove some obsolete files
- pcrexform : checks that we have exactly one substring capture in a signature
- transforms : use generic `Free` functions everywhere

What remains to be done :
- fix `make check` : wrapper to turn `PCRE2_ERROR_UNSET` into an empty string ? or individual changes as in the last commit
- answer TODOs (ie test jit ? keep `PCRE` as suricata feature and add `PCRE2` )
- decide if this is priority in front of https://redmine.openinfosecfoundation.org/issues/4112 https://github.com/OISF/suricata/pull/5966 ie using generic uint parsing and other functional changes to the use pcre1/2

suricata-verify-pr: 480

https://github.com/OISF/suricata-verify/pull/480

Replaces https://github.com/OISF/suricata/pull/6045
- migrate all to pcre2 and remove pcre1
- fix compile warning

Used dummy python script to help get there
```
import sys

f = open(sys.argv[1])
l = f.readline()
while True:
    if "pcre2_substring_copy_bynumber(" in l:
        eol = ';' in l
        while not eol:
            l = l.rstrip() + f.readline().lstrip()
            eol = ';' in l
        b = l[l.find("(")+1:l.rfind(")")]
        a = b.split(",")
        print("pcre2len = %s;" % a[-1])
        a[-2] = "(PCRE2_UCHAR8 *)" + a[-2]
        a = ["parse_regex.match"] + a[3:-1]+ ["&pcre2len"]
        c = l[:l.find("(")+1] + ",".join(a) + l[l.rfind(")"):]
        sys.stdout.write(c)
    else:
        sys.stdout.write(l)
    l = f.readline()
    if l == "":
        break
```
